internal/git/gitsync_processor.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2023 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/numaproj-labs/numaplane/api/v1"
+)
+
+type Message struct {
+	Updates bool
+	Err     error
+}
+
+type GitSyncProcessor struct {
+	gitSync  v1.GitSync
+	k8Client client.Client
+	channels []chan Message
+}
+
+func checkForUpdates(r *git.Repository) (bool, error) {
+	err := r.Fetch(&git.FetchOptions{
+		RemoteName: "origin",
+	})
+	if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
+		return false, err
+	}
+	// Get local and remote references
+	localRef, err := r.Head()
+	if err != nil {
+		return false, err
+	}
+	remoteRef, err := r.Reference(plumbing.NewRemoteReferenceName("origin", "master"), true)
```

Comment:
I saw @chandankumar4 made a comment about this earlier, but I suppose some repos will use "master" and some will use "main" and maybe some could theoretically have a different default branch. Is there a way to determine what the default branch is called?

Reply:
I think this should be the information provided as part of the GitSync CRD. I will create an issue for it.